st7735/lcd_st7735.c

```diff
@@ -307,3 +307,125 @@ Result lcd_st7735_reset(St7735Context *ctx, bool hw) {
 }
 
 Result lcd_st7735_close(St7735Context *ctx) { return (Result){.code = 0}; }
+
+// Determine if an LCD offset of 2 pixels in the narrow dimension and
+// 1 pixel in the wide dimension must be applied for correct function.
+// Involves writing a bundle pixels to the LCD and reading some back
+// from the start of the affected rows to discover the default width.
+//
+// The state of the GM[2:0] config pads of the ST7735 controller within
+// the LCD is the root value we wish to discover, but can only do so
+// by observing side-effects. This function infers GM state from the
+// reset value the of CASET register, which itself must be inferred from
+// pixel buffer behaviour after a reset as it cannot be read directly.
+// The test used is whether the 129th pixel or the 133rd pixel written
+// ends up at the start of the second row, as distinguished by writing
+// different values after the 132nd. We check multiple rows to be sure.
+// Wrapping at 128 infers CASET XE[7:0]=0x7F, which infers GM[2:0]='011'.
+// Wrapping at 132 infers CASET XE[7:0]=0x83, which infers GM[2:0]='000'.
+//
+// GM[2:0]='000' (132x162) is incorrect for the 128x160 panel actually present,
+// meaning minor coordinate offsets are needed. The offsets are 2 pixels
+// in the narrow dimension (x if portrait) and 1 pixel in the wide dimension
+// (y if portrait). This is due to how the ST7735 controller maps the
+// contents of the internal frame buffer to display itself. See the
+// (unfortunately error-ridden) ST7735 datasheet for more details.
+//
+// NOTE1: Must be run after a HW or SW reset and before any CASET commands.
+// NOTE2: Does NOT always perform a reset before returning. State may be dirty.
+// Author: [Elliot Baptist](https://github.com/elliotb-lowrisc)
+
+Result lcd_st7735_check_offset(St7735Context *ctx, size_t *offset_x, size_t *offset_y) {
```

Comment:
Maybe add Elliot as co-author on the commit that adds this check offset function.